server/coordinator/entry_lock.go

```diff
@@ -0,0 +1,62 @@
+package coordinator
+
+import (
+	"sort"
+	"sync"
+)
+
+type EntryLock interface {
+	TryLock(locks []uint64) bool
+	ReleaseLock(locks []uint64) bool
```

Comment:
In `golang` style, `Unlock` seems a better name:
```suggestion
	Unlock(locks []uint64) bool
```
BTW, I guess no need to abstract an interface for it.